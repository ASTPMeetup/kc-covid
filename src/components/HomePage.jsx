import React, { Fragment } from 'react';
import { SocialIcon } from 'react-social-icons';
import logo from '../logo.svg';

// ? Could just combine into home page because i mean it's not much
const HomeHeader = () => (
  <div className="homepage-header">
    <SocialIcon url="https://twitter.com/QuintonLucasKC" />
    <h3>Some text to put in the header</h3>
    <SocialIcon url="https://twitter.com/QuintonLucasKC" style={{ height: 25, width: 25 }} />
  </div>
);

// TODO make home page
// TODO add social accounts
function HomePage() {
  // SocialIcon automatically pulls the icon from the url or you can specify it
  return (
    <Fragment>
      <HomeHeader />
      <div className="App-header">
        <img src={logo} className="App-logo" alt="logo" />
        <a className="App-link" href="https://reactjs.org" target="_blank" rel="noopener noreferrer">
          Learn React
        </a>
      </div>
    </Fragment>
  );
}

export default HomePage;
