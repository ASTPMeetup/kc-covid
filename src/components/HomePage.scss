.social-links {
  display: inline;
}

.social-icon {
  margin: 5px;
}
