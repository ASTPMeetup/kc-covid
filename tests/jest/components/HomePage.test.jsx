import React from 'react';
import { shallow } from 'enzyme';
import HomePage from '../../../src/components/HomePage';

describe('HomePage', () => {
  it('renders a the home page view', () => {
    const homePage = shallow(<HomePage />);
    expect(homePage).toMatchSnapshot();
  });
});
