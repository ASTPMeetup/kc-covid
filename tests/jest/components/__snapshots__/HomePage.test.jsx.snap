// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage renders a the home page view 1`] = `
<Fragment>
  <HomeHeader />
  <div
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src={null}
    />
    <a
      className="App-link"
      href="https://reactjs.org"
      rel="noopener noreferrer"
      target="_blank"
    >
      Learn React
    </a>
  </div>
</Fragment>
`;
